packages/client/src/admin/components/Logs.tsx

```diff
@@ -14,20 +18,97 @@ const Container = styled.div`
   overflow: scroll;
 `;
 
+const ScrollDownButton = styled.button`
+  position: absolute;
+  bottom: 10px;
+  right: 30px;
+  border: none;
+  border-radius: 0;
+  min-height: 23px;
+  min-width: 75px;
+  padding: 0 12px;
+  background: silver;
+  box-shadow: inset -1px -1px #0a0a0a, inset 1px 1px #fff, inset -2px -2px grey, inset 2px 2px #dfdfdf;
+  font-family: sans-serif;
+  font-size: 13px;
+
+  &:active {
+    box-shadow: inset -1px -1px #fff, inset 1px 1px #0a0a0a, inset -2px -2px #dfdfdf, inset 2px 2px grey;
+    padding: 2px 11px 0 13px;
+  }
+`;
+
 interface LogsProps {
   logs: Log[],
 }
 
 function Logs({ logs }: LogsProps) {
+  const containerRef = React.useRef<HTMLDivElement>(null);
+  const [updateScroll, setUpdateScroll] = React.useState<boolean>(false);
+  const [showScrollDownButton, setShowScrollDownButton] = React.useState<boolean>(true);
+
+  const scrollDown = (): void => {
+    const container = containerRef.current;
+    if (container) {
+      container.scrollTo({
+        left: 0,
+        top: container.scrollHeight,
+        behavior: 'smooth',
+      });
+    }
+  };
+
+  const isScrolledToBottom = (): boolean => {
+    if (!containerRef.current) {
+      return false;
+    }
+    const {
+      scrollHeight,
+      scrollTop,
+      clientHeight,
+    } = containerRef.current;
+
+    return scrollHeight - scrollTop - clientHeight < 25;
+  };
+
+  const autoScrollToBottom = (): void => {
+    const scrolledToBottom = isScrolledToBottom();
+
+    if (updateScroll && !scrolledToBottom) {
+      setUpdateScroll(false);
+      setShowScrollDownButton(true);
+      return;
+    }
+
+    if (!updateScroll && scrolledToBottom) {
+      setUpdateScroll(true);
+      setShowScrollDownButton(false);
+    }
+  };
+
+  React.useEffect(() => {
+    if (updateScroll) {
+      scrollDown();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Ja bym chyba jednak dał pełną listę

Reply:
Tutaj bardziej chodzi o to, że jak dodam `updateScroll` do listy to się wykona przeskrolowanie automatyczne jak będzie wystarczająco nisko przeskrolowany użytkownik, a to powinno się uruchamiać tylko wtedy kiedy wpada nowy wpis do logów.